addons/google_calendar/static/tests/google_calendar_tests.js

```diff
@@ -152,4 +153,64 @@ QUnit.module('Google Calendar', {
 
         assert.containsN(target, '.fc-event-container', 6, "should now display 6 events on the month");
     });
+
+    QUnit.test("component is destroyed while sync google calendar", async function (assert) {
+        assert.expect(4);
+        const def = makeDeferred();
+        serverData.actions = {
+            1: {
+                id: 1,
+                name: "Partners",
+                res_model: "calendar.event",
+                type: "ir.actions.act_window",
+                views: [
+                    [false, "list"],
+                    [false, "calendar"],
+                ],
+            },
+        };
+
+        serverData.views = {
+            "calendar.event,false,calendar": `
+                <calendar class="o_calendar_test" js_class="attendee_calendar" date_start="start" date_stop="stop">
+                    <field name="name"/>
+                    <field name="partner_ids" write_model="filter_partner" write_field="partner_id"/>
+                </calendar>`,
+            "calendar.event,false,list": `<tree sample="1"/>`,
+            "calendar.event,false,search": `<search />`,
+        };
+
+        const webClient = await createWebClient({
+            serverData,
+            async mockRPC(route, args) {
+                if (route === '/google_calendar/sync_data') {
+                    assert.step(route);
+                    return def;
+                } else if (route === '/web/dataset/call_kw/calendar.event/search_read') {
+                    assert.step(route);
+                } else if (route === '/web/dataset/call_kw/res.partner/get_attendee_detail') {
+                    return Promise.resolve([]);
+                } else if (route === '/web/dataset/call_kw/res.users/has_group') {
+                    return Promise.resolve(true);
+                }
+            },
+        });
+
+        await doAction(webClient, 1);
+
+        click(target.querySelector(".o_cp_switch_buttons .o_calendar"));
+        await nextTick();
+
+        click(target.querySelector(".o_cp_switch_buttons .o_calendar"));
+        await nextTick();
+
+        def.resolve();
```

Comment:
don't you instead need to wait for one second to reproduce the crash? I didn't run the test, but if I had to guess I would say it doesn't fail without the fix.

Reply:
Yes, we need to wait locally. We can reproduce the issue directly on the runbot because the process is already slowed down. The test is also failing without the fix, and we get the error that the component is destroyed.

Reply:
Mmmmh we don't correctly mock the rpc service, that's why you're able to reproduce the issue that way. But in the "real" environment, the issue occurs because of the `new Promise(resolve => setTimeout(resolve, 1000)),` getting resolved first. In master (with hoot), you might need to call `runAllTimers` (to simulate that the timeout is over) instead of resolving the promise (which is the exact scenario of the issue).